### DISCUSSION TITLE
problem to access pthread during build (goal is to link mumps and mfem)

### DISCUSSION THREAD

**Discussion by sp3c333** [scivision/mumps-superbuild#54] [Q&A] [answered]
Hello,

Here is what I am trying to accomplish : 
I am trying to use mumps on mfem and I am using mfem in visual studio 2022 (I am on windows 11). For now I do not need the parallel version of mumps. 

Here is what I did : 
I used cmake with ninja as generator to configure mumps with these options :
The following features have been enabled:
 * 64-bit-integer, use 64-bit integers in C and Fortran
 * real32, Build with single precision
 * real64, Build with double precision
 * complex32, Build with complex precision
 * complex64, Build with complex16 precision
 * 
-- The following features have been disabled:

 * Parallel, parallel MUMPS (using MPI and Scalapack)
 * GEMMT, use GEMMT for symmetric matrix-matrix multiplication
 * Scotch, Scotch graph partitioning https://www.labri.fr/perso/pelegrin/scotch/
 * Openmp, OpenMP API https://www.openmp.org/
 * shared, Build shared libraries
Configuring done
Generating done

When I try to build using the command "cmake --build build" in "Intel oneAPI command prompt for Intel 64 for Visual Studio", I encountered an error saying that my pthread.h file could not be found so I downloaded "pthread-win32-master" and I created an include directory in pthread where I put my .h files and in the cmakelists.txt file of mumps I added this line : include_directories(C:/Users/XXXX/MFEM_Windows/mfem-4.5-new/pthread-win32-master/include)
It solved this error

I successfully builded mumps but blocked at the tests where I get error messages for all 4 tests saying : 
c:/users/XXXX/gcc/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: libmumps_common.a(mumps_io_err.c.obj):mumps_io_err.c:(.text+0x6e): undefined reference to `pthread_mutex_init'

I have not been able to solve this error message. I have tried to add these lines in the cmakelists.txt of the test directory : 
target_link_libraries(mumpscfg PRIVATE C:/Users/XXXX/MFEM_Windows/mfem-4.5-new/pthread-win32-master/build/Release/pthreadVC3.lib)
without success

Could you help me please ?

**Comment by sp3c333** [comment, marked as answer]
I found my mistake: I used cmake gui for the configure and generate steps while what should be done is : "cmake -G Ninja -B buildTT -DBUILD_SINGLE=yes -DBUILD_DOUBLE=yes -DBUILD_COMPLEX=yes -DBUILD_COMPLEX16=yes -Dparallel=no -Dintsize64=yes -Dgemmt=no" in the intel prompt.

However I have difficulties linking mumps and mfem. I am not sure if mfem can handle mumps serial. Do you have any advice on how to use mumps serial with mfem in visual studio 2022 on windows 11 ?

**Reply by scivision** [reply]
I would avoid using CMake-GUI yes. Maybe MFEM or MUMPS users email list has someone with the experience.
